src/components/context/utilities.js

```diff
@@ -53,6 +53,14 @@ export const getAppDataByQuestionId = (appData, qId) => {
   );
 };
 
+export const getAllAppDataByQuestionId = (appData, qId) => {
+  return (
+    appData?.filter(({ data }) => data?.questionId === qId) ?? [
+      { data: { questionId: qId } },
+    ]
+  );
```

Comment:
I think it's a bit counter-intuitive to return a truthy value. I think the default value should be set a bit higher in the logic.

Reply:
I totally agree with you, I actually did that because the function above (`getAppDataByQuestionId`) was doing that, so to be consistent I thought I should do the same. 

I think if the `appData` is `null` or `undefined` I should simply return an empty array. What do you think about that ?